doubtlab/reason.py

```diff
@@ -72,6 +72,44 @@ def __call__(self, X, y=None):
         return np.where(rvals < self.probability, rvals, 0)
 
 
+class ShannonEntropyReason:
+    """
+    Assign doubt when the normalized Shannon entropy is too high.
+
+    Arguments:
+        model: scikit-learn classifier
+        threshold: confidence threshold for doubt assignment
+
+    Usage:
+
+    ```python
+    from sklearn.datasets import load_iris
+    from sklearn.linear_model import LogisticRegression
+
+    from doubtlab.ensemble import DoubtEnsemble
+    from doubtlab.reason import ShannonEntropyReason
+
+    X, y = load_iris(return_X_y=True)
+    model = LogisticRegression(max_iter=1_000)
+    model.fit(X, y)
+
+    doubt = DoubtEnsemble(reason = ShannonEntropyReason(model=model))
+
+    indices = doubt.get_indices(X, y)
+    ```
+    """
+
+    def __init__(self, model, threshold=0.5):
+        self.model = model
+        self.threshold = threshold
+
+    def __call__(self, X, y):
```

Comment:
Could you add a `from_proba` method to this class? That way, you might be able to re-use probas without re-computing. There's a PR in the works for the other reasons [here](https://github.com/koaning/doubtlab/pull/23/files#diff-d954109e3f3329a16a18bee13ef3dd03a107a331e19cc7911fd0a6a041421e16R40-R44). 

Having such a method will also make it easier to add a unit test, which is currently missing in this PR.

Reply:
Done!